### PR TITLE
MGMT-10121: Enhance logging for package proxy

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -41,12 +41,12 @@ func (p *proxy) Setup(ctx context.Context, obj *unstructured.Unstructured) error
 
 	if strings.Compare(obj.GetKind(), "Pod") == 0 {
 		if err := p.setupPod(ctx, obj); err != nil {
-			return errors.Wrap(err, "Cannot setup Pod Proxy")
+			return fmt.Errorf("failed to setup Pod %s/%s proxy: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/proxy/proxy.go:44]:
```suggestion
			return fmt.Errorf("failed to setup pod/%s proxy in namespace %s: %w", obj.GetName(), obj.GetNamespace(), err)
```
WDYT?

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -41,12 +41,12 @@ func (p *proxy) Setup(ctx context.Context, obj *unstructured.Unstructured) error
...
 		}
 	}
 	if strings.Compare(obj.GetKind(), "DaemonSet") == 0 {
 		if err := p.setupDaemonSet(ctx, obj); err != nil {
-			return errors.Wrap(err, "Cannot setup DaemonSet Proxy")
+			return fmt.Errorf("failed to setup DaemonSet %s/%s proxy: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/proxy/proxy.go:49]: Same as https://github.com/openshift/special-resource-operator/pull/195/files#r870066310

File: pkg/proxy/proxy.go [hunk 3]
```diff
@@ -59,15 +59,15 @@ func (p *proxy) Setup(ctx context.Context, obj *unstructured.Unstructured) error
 func (p *proxy) setupDaemonSet(ctx context.Context, obj *unstructured.Unstructured) error {
 	containers, found, err := unstructured.NestedSlice(obj.Object, "spec", "template", "spec", "containers")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to find slice spec/template/spec/containers in daemon object %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/proxy/proxy.go:62]:
```suggestion
		return fmt.Errorf("failed to find slice spec.template.spec.containers in daemonset/%s in namespace %s: %w", obj.GetName(), obj.GetNamespace(), err)
```

File: pkg/proxy/proxy.go [hunk 4]
```diff
@@ -59,15 +59,15 @@ func (p *proxy) Setup(ctx context.Context, obj *unstructured.Unstructured) error
...
 	}
 
 	if !found {
-		return fmt.Errorf("spec.template.spec.containers not found in the daemon yaml")
+		return fmt.Errorf("spec.template.spec.containers not found in the daemon object %s/%s", obj.GetNamespace(), obj.GetName())
```
Comment (ybettan) [inline pkg/proxy/proxy.go:66]:
```suggestion
		return fmt.Errorf("spec.template.spec.containers not found in the daemonset/%s in namespace %s", obj.GetName(), obj.GetNamespace())
```

File: pkg/proxy/proxy.go [hunk 5]
```diff
@@ -59,15 +59,15 @@ func (p *proxy) Setup(ctx context.Context, obj *unstructured.Unstructured) error
...
 	}
 
 	if err = p.setupContainersProxy(ctx, containers); err != nil {
-		return fmt.Errorf("cannot set proxy for Pod: %w", err)
+		return fmt.Errorf("failed to set proxy for containers of the DaemonSet %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/proxy/proxy.go:70]:
```suggestion
		return fmt.Errorf("failed to set proxy for containers of daemonset/%s in namespace %s: %w", obj.GetName(), obj.GetNamespace(), err)
```

File: pkg/proxy/proxy.go [hunk 6]
```diff
@@ -76,11 +76,11 @@ func (p *proxy) setupDaemonSet(ctx context.Context, obj *unstructured.Unstructur
 func (p *proxy) setupPod(ctx context.Context, obj *unstructured.Unstructured) error {
 	containers, found, err := unstructured.NestedSlice(obj.Object, "spec", "containers")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to find slice spec/containers in Pod object %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/proxy/proxy.go:79]:
```suggestion
		return fmt.Errorf("failed to find slice spec.containers in pod/%s in namespace %s: %w", obj.GetName(), obj.GetNamespace(), err)
```

File: pkg/proxy/proxy.go [hunk 7]
```diff
@@ -76,11 +76,11 @@ func (p *proxy) setupDaemonSet(ctx context.Context, obj *unstructured.Unstructur
...
 	}
 
 	if !found {
-		return fmt.Errorf("spec.containers not found in the pod yaml")
+		return fmt.Errorf("spec.containers not found in the pod object %s/%s", obj.GetNamespace(), obj.GetName())
```
Comment (ybettan) [inline pkg/proxy/proxy.go:83]:
```suggestion
		return fmt.Errorf("spec.containers not found in pod/%s in namespace %s", obj.GetName(), obj.GetNamespace())
```

File: pkg/proxy/proxy.go [hunk 8]
```diff
@@ -97,7 +97,7 @@ func (p *proxy) setupContainersProxy(ctx context.Context, containers []interface
 		case map[string]interface{}:
 			env, found, err := unstructured.NestedSlice(container, "env")
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to get slice evn from container: %w", err)
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:100]: fixed

File: pkg/proxy/proxy.go [hunk 9]
```diff
@@ -142,10 +142,10 @@ func (p *proxy) ClusterConfiguration(ctx context.Context) (Configuration, error)
 
 	proxiesAvailable, err := p.kubeClient.HasResource(configv1.SchemeGroupVersion.WithResource("proxies"))
 	if err != nil {
-		return *proxy, errors.Wrap(err, "Error discovering proxies API resource")
+		return *proxy, fmt.Errorf("failed to discover cluster resources proxies: %w", err)
 	}
 	if !proxiesAvailable {
-		log.Info("Warning: Could not find proxies API resource. Can be ignored on vanilla K8s.")
+		log.Info(utils.WarnString("could not find proxies API resource. Can be ignored on vanilla K8s."))
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:148]: fixed

File: pkg/proxy/proxy.go [hunk 10]
```diff
@@ -166,22 +166,22 @@ func (p *proxy) ClusterConfiguration(ctx context.Context) (Configuration, error)
 		// and initialized the Proxy struct with zero sized strings
 		if strings.Contains(cfgName, "cluster") {
 			if proxy.HttpProxy, _, err = unstructured.NestedString(cfg.Object, "spec", "httpProxy"); err != nil {
-				log.Error(err, "Failed to obtain httpProxy")
+				log.Info(utils.WarnString("failed to obtain httpProxy"), "namespace", cfg.GetNamespace(), "name", cfg.GetName())
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:169]: fixed

File: pkg/proxy/proxy.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:44]: i prefer the annotation namespace/name. It is more in line with kubectl, and this is how we use it in other packages also

File: pkg/proxy/proxy.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:49]: i prefer the annotation namespace/name. It is more in line with kubectl, and this is how we use it in other packages also

File: pkg/proxy/proxy.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:62]: changed daemon object to daemonset, but keeping the namespace/name annotation

File: pkg/proxy/proxy.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:66]: changed daemon object to daemonset, but keeping the namespace/name annotation

File: pkg/proxy/proxy.go (hunk 1 above)
Comment (ybettan) [reply pkg/proxy/proxy.go:44]: In k8s it is used `type/objname` or `type objname`.
We also have packages in which we use `kind/objname` so we will probably won't be consistent across all the project then.

In that case, I don't mind keeping it.

File: pkg/proxy/proxy.go (hunk 6 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:79]: changed daemon object to pod, but keeping the namespace/name annotation

File: pkg/proxy/proxy.go (hunk 7 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:83]: changed daemon object to pod, but keeping the namespace/name annotation

File: pkg/proxy/proxy.go (hunk 3 above)
Comment (ybettan) [reply pkg/proxy/proxy.go]: How about replacing `spec/template/spec/containers` with `spec.template.spec.containers`?

File: pkg/proxy/proxy.go (hunk 6 above)
Comment (ybettan) [reply pkg/proxy/proxy.go]: How about changing `spec/containers` to `spec.containers`?

File: pkg/proxy/proxy.go [hunk 11]
```diff
@@ -59,15 +59,15 @@ func (p *proxy) Setup(ctx context.Context, obj *unstructured.Unstructured) error
 func (p *proxy) setupDaemonSet(ctx context.Context, obj *unstructured.Unstructured) error {
 	containers, found, err := unstructured.NestedSlice(obj.Object, "spec", "template", "spec", "containers")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to find slice spec/template/spec/containers in daemonset %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/proxy/proxy.go:62]:
```suggestion
		return fmt.Errorf("failed to find slice spec.template.spec.containers in daemonset %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```

Don't you prefer it? we could confuse it with `namespace/name` and also, this is how I did it in some pakcages.

File: pkg/proxy/proxy.go (hunk 2 above)
Comment (ybettan) [reply pkg/proxy/proxy.go:49]: I don't mind then.

File: pkg/proxy/proxy.go (hunk 11 above)
Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy.go:62]: fixed

**Pull request merged**